linux/LinuxProcess.c

```diff
@@ -195,6 +204,303 @@ void LinuxProcess_printDelay(float delay_percent, char* buffer, int n) {
 }
 #endif
 
+/*
+Note that when colorizing a basename with the comm prefix, the entire basename
+(not just the comm prefix) is colorized for better readability, and it is
+implicit that only upto (TASK_COMM_LEN - 1) could be comm
+*/
+static inline bool findCommInCmdline(const char *comm, char *cmdline, int cmdlineBasenameOffset, int *pCommStart, int *pCommEnd) {
+   /* Try to find procComm in tokenized cmdline - this might in rare cases
+    * mis-identify a string or fail, if comm or cmdline had been unsuitably
+    * modified by the process */
+   bool lastToken = false;
+   char *token;
+   char *tokenBase;
+
+   for (token = cmdline + cmdlineBasenameOffset; *token; ) {
+      for (tokenBase = token; *token && *token != '\n'; ++token) {
+         if (*token == '/') {
+            tokenBase = token + 1;
+         }
+      }
+      if (*token) {  /* NUL terminate at delimiter for strncmp */
+         *token = 0;
+      } else {
+         lastToken = true;
+      }
+
+      if (strncmp(tokenBase, comm, TASK_COMM_LEN - 1) == 0) {
+         if (!lastToken) { /* restore the delimiter */
+            *token = '\n';
+         }
+         *pCommStart = tokenBase - cmdline;
+         *pCommEnd = token - cmdline;
+         return true;
+      }
+      if (!lastToken) {
+         /* restore the delimiter and skip repeats */
+         *token = '\n';
+         do {
+            ++token;
+         } while ('\n' == *token);
+      }
+   }
+   return false;
+}
+
+static inline int matchCmdlinePrefixWithExeSuffix(const char *cmdline, int cmdlineBaseOffset, const char *exe, int exeBaseOffset, int exeBaseLen) {
+   int matchLen;       /* matching length to be returned */
+   char delim;         /* delimiter following basename */
+
+   /* cmdline prefix is an absolute path: it must match whole exe. */
+   if (cmdline[0] == '/') {
+      matchLen = exeBaseLen + exeBaseOffset;
+      if (strncmp(cmdline, exe, matchLen) == 0 &&
+          ((delim = cmdline[matchLen]) == 0 || delim == '\n' || delim == ' ')) {
```

Comment:
While technically correct this if statement might be better of broken down into two checks avoiding the inline assignment …

Reply:
Agreed